### PR TITLE
Start of demo server

### PR DESCRIPTION
**Pull request by jimmarino**

### CODE REVIEW

File: extensions/cloud/portability-cloud-local/src/main/java/org/dataportabilityproject/cloud/local/LocalJobStore.java [hunk 1]
```diff
@@ -16,21 +16,30 @@
...
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
-import org.dataportabilityproject.spi.cloud.storage.JobStore;
-import org.dataportabilityproject.spi.cloud.types.JobAuthorization;
-import org.dataportabilityproject.spi.cloud.types.PortabilityJob;
 
 /** An in-memory {@link JobStore} implementation that uses a concurrent map as its store. */
 public final class LocalJobStore implements JobStore {
-  private final ConcurrentHashMap<UUID, Map<String, Object>> map;
+  private static ConcurrentHashMap<UUID, Map<String, Object>> SINGLETON_MAP;
```
Comment (rtannenbaum) [inline extensions/cloud/portability-cloud-local/src/main/java/org/dataportabilityproject/cloud/local/LocalJobStore.java]: As discussed on the call, I think you can assume that if this store is being used, we want the singleton

The other local impl uses GoogleJobStore backed by DataStore local emulator

I think we can get rid of the singleVM flag, and instead have LocalCloudExtension use the 'cloud' and 'env' flags to determine whether to return LocalJobStore or one of the emulator stores (I can do that along with #234)

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java [hunk 2]
```diff
@@ -41,16 +58,119 @@ public void uncaughtException(Thread thread, Throwable t) {
...
     // TODO: Don't hardcode list of services
     Set<String> services = ImmutableSet.of("microsoft");
 
+    // FIXME make configurable
+    SymmetricKeyGenerator keyGenerator = new AesSymmetricKeyGenerator();
+
     Injector injector =
         Guice.createInjector(
-            new PortabilityAuthServiceProviderModule(services), new ReferenceApiModule());
+            new ApiServicesModule(typeManager, cloudExtension.getJobStore(), keyGenerator),
```
Comment (rtannenbaum) [inline portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java]: this guice config looks off to me, but if it gets stuff working for the demo we can look into it more later

if you look at WorkerMain that is all configured to use Guice, you should only need to install the ApiModule and everything else should be automagically configured from there..

File: extensions/cloud/portability-cloud-local/src/main/java/org/dataportabilityproject/cloud/local/LocalCloudExtension.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 Google Inc.
```
Comment (seehamrun) [inline extensions/cloud/portability-cloud-local/src/main/java/org/dataportabilityproject/cloud/local/LocalCloudExtension.java]: nit: update this

Comment (jimmarino) [reply extensions/cloud/portability-cloud-local/src/main/java/org/dataportabilityproject/cloud/local/LocalCloudExtension.java]: It looks like there are a bunch of these. I've added an issue 241 for this.

File: extensions/cloud/portability-cloud-local/src/main/java/org/dataportabilityproject/cloud/local/LocalJobStore.java (hunk 1 above)
Comment (jimmarino) [reply extensions/cloud/portability-cloud-local/src/main/java/org/dataportabilityproject/cloud/local/LocalJobStore.java]: I removed the singleVM flag. If you can re-work it to use the other flags in 234 that would be great.

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java (hunk 2 above)
Comment (jimmarino) [reply portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java]: I added the ApiServicesModule to make the instances created at bootstrap available for injection in core services. Is there something that doesn't seem right about that?

Comment (rtannenbaum) [reply portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java]: actually never mind, I forgot that JobStore etc isn't provided by Guice all the time, only for CloudExtensions who choose to use Guice. You can disregard this...

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java [hunk 4]
```diff
@@ -41,16 +58,119 @@ public void uncaughtException(Thread thread, Throwable t) {
...
+    ApiMain apiMain = new ApiMain();
+    apiMain.initialize();
+    apiMain.start();
+  }
+
+  public void initialize() {
+    TypeManager typeManager = new TypeManagerImpl();
+
+    // TODO implement
+    Map<String, Object> configuration = new HashMap<>();
```
Comment (rtannenbaum) [inline portability-gateway/src/main/java/org/dataportabilityproject/gateway/ApiMain.java]: FYI feel free to use the config parsing stuff from WorkerExtensionContext I added, if it'd be helpful

String tempSettings = "environment: LOCAL\ncloud: LOCAL";
InputStream in = new ByteArrayInputStream(tempSettings.getBytes(StandardCharsets.UTF_8));
config = ConfigUtils.parse(in);

**Comment by jimmarino** [comment]
Yikes, I messed up the rebase. Let me revert and fix....

**Comment by jimmarino** [comment]
OK fixed it.

**Pull request merged**